File: src/main/java/org/axostudio/axohologram/animation/AnimationSettings.java
package org.axostudio.axohologram.animation;

public record AnimationSettings(
        boolean enabled,
        long tickRate,
        boolean cacheFrames,
        boolean asyncTextProcessing,
        boolean allowPlaceholdersInsideAnimations,
        boolean reduceQualityOnLowTps,
        boolean displayInterpolation
) {

    public AnimationSettings {
        tickRate = Math.max(1L, tickRate);
    }

    public static AnimationSettings defaults() {
        return new AnimationSettings(true, 2L, true, true, false, true, true);
    }

    public int interpolationDuration() {
        return displayInterpolation ? (int) Math.min(Math.max(tickRate, 1L), 20L) : 0;
    }
}

File: src/main/kotlin/org/axostudio/axohologram/feature/animation/AnimationSettings.kt
package org.axostudio.axohologram.feature.animation

data class AnimationSettings(
    val enabled: Boolean = true,
    val tickRate: Long = 2L,
    val cacheFrames: Boolean = true,
    val asyncTextProcessing: Boolean = true,
    val displayInterpolation: Boolean = true
) {
    val interpolationDuration: Int
        get() = if (displayInterpolation) tickRate.coerceIn(1L, 20L).toInt() else 0

    companion object {
        fun defaults(): AnimationSettings = AnimationSettings()
    }
}
